roles/accprovision/tasks/main.yml

```diff
@@ -1,61 +1,63 @@
-  stat:
-    path: "/tmp/acc_provision/version"
-  register: file_present
-
-- name: Get current resourceVersion if version file exists
+- name: Get acc-provision configmap
+  k8s_info:
+    api_version: v1
+    kind: ConfigMap
+    name: acc-provision-config
+    namespace: "{{ lookup('env', 'WATCH_NAMESPACE') }}"
+  register: cmap_data
+
+- name: Collecting config-map content
   set_fact:
-    old_version: "{{ lookup('file', '/tmp/acc_provision/version') }}"
-  when: file_present.stat.exists
+    cmap_content: "{{ cmap_data['resources'][0]}}"
 
-- name: Create resourceVersion file if it doesn't exist
-  file:
-    path: /tmp/acc_provision/version
-    state: touch
-  when: not file_present.stat.exists
+- name: Write cmap YAML
+  copy:
+    dest: "{{ acc_provision_dir_path }}/cmap.yaml"
+    content: "{{ cmap_content }}"
 
-- name: Set old resourceVersion to empty string if no cached value exists
+- name: Collecting config-map spec
   set_fact:
-    old_version: ""
-  when: not file_present.stat.exists
+    cmap_input: "{{ cmap_data['resources'][0]['data']['spec'] }}"
 
-- name: Old resourceVersion value
-  debug:
-    msg: "{{ old_version }}"
+- name: Get CRD definition for accprovisioninput
+  community.kubernetes.k8s_info:
+    api_version: v1
+    kind: CustomResourceDefinition
+    name: accprovisioninputs.aci.ctrl
+  register: raw_crd
 
-- name: Get required accprovisioninput CR name
+# This task picks the first schema defined under the CRD
+- name: Get crd definition from raw k8s resource
   set_fact:
-    acc_provision_input_cr_name: "{{ lookup('env', 'ACC_PROVISION_INPUT_CR_NAME') }}"
-
-- name: Cache the new resourceVersion
-  copy: content="{{ cr_version }}" dest='/tmp/acc_provision/version'
-  when: cr_name == acc_provision_input_cr_name
-
-- name: Upgrade cluster
-  include_tasks: process.yaml
-  when: cr_name == acc_provision_input_cr_name and cr_version != old_version
+    crd_definition: "{{ raw_crd['resources'][0]['spec']['versions'][0]['schema']['openAPIV3Schema']['properties']['spec']['properties']['acc_provision_input']['properties'] }}"
+
+- name: Compute acc-provision input file
+  script: "scripts/compute_input_yaml.py"
+  args:
+    executable: /usr/bin/python3
+  environment:
+    CRDINPUT: "{{ acc_provision_input }}"
+    CRD_DEFINITION: "{{ crd_definition }}"
+    CONFIGMAP_INPUT_FILE: "{{ acc_provision_dir_path }}/cmap.yaml"
+    ACCPROVDIR: "{{ acc_provision_dir_path }}"
+    ACCPROVFILE: "{{ acc_provision_file_name }}"
+
+- name: Generate new deployment yaml
+  shell: "acc-provision -c {{ acc_provision_file_name }} -f {{ lookup('env', 'ACC_PROVISION_FLAVOR') }} -o {{ aci_cni_deployment_file }} --operator-mode True"
+  args:
+    chdir: "{{ acc_provision_dir_path }}"
+
+- name: Run k8s command to apply deployment file
+  k8s:
+    state: present
+    src: "{{ acc_provision_dir_path }}/{{ aci_cni_deployment_file }}"
+  register: k8s_result
+  until: k8s_result is not failed
+  retries: 5
+  delay: 30
```

Comment:
is this reconcile also has to be 60 seconds with 5 retries similar to reconcilePeriod in watches.yaml. Or is 30 sec is sufficient?

Reply:
This is just to retry this k8s call- the reconcilePeriod is to run the entire Ansible role every 60s.

Reply:
So this timeout is sufficient [Tested]. Just confirming